services/src/main/java/org/keycloak/userprofile/config/UPConfigUtils.java

```diff
@@ -127,6 +130,25 @@ private static void validate(KeycloakSession session, UPAttribute attributeConfi
         }
         if (attributeConfig.getRequired() != null) {
             validateRoles(attributeConfig.getRequired().getRoles(), "required.roles", errors, attributeName);
+            validateScopes(attributeConfig.getRequired().getScopes(), "required.scopes", attributeName, errors, session);
+        }
+        if (attributeConfig.getSelector() != null) {
+            validateScopes(attributeConfig.getSelector().getScopes(), "selector.scopes", attributeName, errors, session);
+        }
+    }
+
+    private static void validateScopes(Set<String> scopes, String propertyName, String attributeName, List<String> errors, KeycloakSession session) {
+        if (scopes == null) {
+            return;
+        }
+
+        for (String scope : scopes) {
+            RealmModel realm = session.getContext().getRealm();
+            Stream<ClientScopeModel> realmScopes = realm.getClientScopesStream();
+
+            if (!realmScopes.anyMatch(cs -> cs.getName().equals(scope))) {
+                errors.add(new StringBuilder("'").append(propertyName).append("' configuration for attribute '").append(attributeName).append("' contains unsupported role '").append(scope).append("'").toString());
```

Comment:
Ooops, looks like typo there `contains unsupported role` should be `contains unsupported scope`

Reply:
Fixed